pkg/services/ngalert/notifier/channels/pushover.go

```diff
@@ -32,112 +32,98 @@ var (
 // alert notifications to Pushover
 type PushoverNotifier struct {
 	*Base
-	UserKey          string
-	APIToken         string
-	AlertingPriority int
-	OKPriority       int
-	Retry            int
-	Expire           int
-	Device           string
-	AlertingSound    string
-	OKSound          string
-	Upload           bool
-	Message          string
-	tmpl             *template.Template
-	log              log.Logger
-	images           ImageStore
-	ns               notifications.WebhookSender
+	tmpl     *template.Template
+	log      log.Logger
+	images   ImageStore
+	ns       notifications.WebhookSender
+	settings pushoverSettings
 }
 
-type PushoverConfig struct {
+type pushoverSettings struct {
 	*NotificationChannelConfig
-	UserKey          string
-	APIToken         string
-	AlertingPriority int
-	OKPriority       int
-	Retry            int
-	Expire           int
-	Device           string
-	AlertingSound    string
-	OKSound          string
-	Upload           bool
-	Message          string
+	UserKey          string `json:"userKey,omitempty" yaml:"userKey,omitempty"`
+	APIToken         string `json:"apiToken,omitempty" yaml:"apiToken,omitempty"`
+	alertingPriority int
+	okPriority       int
+	retry            int
+	expire           int
+	Device           string `json:"device,omitempty" yaml:"device,omitempty"`
+	AlertingSound    string `json:"sound,omitempty" yaml:"sound,omitempty"`
+	OKSound          string `json:"okSound,omitempty" yaml:"okSound,omitempty"`
+	upload           bool
+	Title            string `json:"title,omitempty" yaml:"title,omitempty"`
+	Message          string `json:"message,omitempty" yaml:"message,omitempty"`
 }
 
 func PushoverFactory(fc FactoryConfig) (NotificationChannel, error) {
-	cfg, err := NewPushoverConfig(fc.Config, fc.DecryptFunc)
+	pn, err := newPushoverNotifier(fc)
 	if err != nil {
 		return nil, receiverInitError{
 			Reason: err.Error(),
 			Cfg:    *fc.Config,
 		}
 	}
-	return NewPushoverNotifier(cfg, fc.ImageStore, fc.NotificationService, fc.Template), nil
+	return pn, nil
 }
 
-func NewPushoverConfig(config *NotificationChannelConfig, decryptFunc GetDecryptedValueFn) (*PushoverConfig, error) {
-	userKey := decryptFunc(context.Background(), config.SecureSettings, "userKey", config.Settings.Get("userKey").MustString())
-	if userKey == "" {
+// newSlackNotifier is the constructor for the Slack notifier
```

Comment:
```suggestion
// newPushoverNotifier is the constructor for the Pushover notifier
```

Reply:
You're right, I'll do the title directly. But we should come to an agreement on whether to use `simplejson` or `encoding/json`.